### PR TITLE
:bug: Fixed ``design_sidb_gates`` function for skeletons with east-facing I/O pins.

### PR DESCRIPTION
**Pull request by Drewniok**
## Description

This PR fixes the following bug in the ``design_sidb_gates`` function: When using a gate skeleton with wires facing east, the gate design would not work. 

The error occurred during the second pruning step when using _QuickCell_ for designing SiDB gates. During this step, the feasibility of the physical validity is checked by toggling through different input patterns and adjusting the charge distribution of the I/O pins accordingly. However, the logic failed for I/O pins facing east because this specific case was not accounted for in the if statement. This is now fixed.
 

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [x] I have added a changelog entry.
- [x] I have created/adjusted the Python bindings for any new or updated functionality.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -33,6 +33,7 @@ Changed
 
 Fixed
 #####
+- Fixed a bug in the gate design when using skeletons with I/O wires facing east.
```
Comment (marcelwa) [inline docs/changelog.rst:36]:
```suggestion
- Fixed a bug in SiDB gate design when using skeletons with I/O wires facing east
```

File: include/fiction/algorithms/physical_design/design_sidb_gates.hpp [hunk 2]
```diff
@@ -366,11 +384,18 @@ class design_sidb_gates_impl
...
     {
         mockturtle::stopwatch stop{stats.time_total};
         const auto            gate_candidates = run_pruning();
 
+        stats.number_of_layouts_after_first_pruning =
+            all_canvas_layouts.size() - number_of_discarded_layouts_at_first_pruning;
+        stats.number_of_layouts_after_second_pruning =
+            stats.number_of_layouts_after_first_pruning - number_of_discarded_layouts_at_second_pruning;
+        stats.number_of_layouts_after_third_pruning =
+            stats.number_of_layouts_after_second_pruning - number_of_discarded_layouts_at_third_pruning;
```
Comment (marcelwa) [inline include/fiction/algorithms/physical_design/design_sidb_gates.hpp:397]: To atomically obtain the values stored in `std::atomic`, you should use `.load()`.

https://en.cppreference.com/w/cpp/atomic/atomic

File: include/fiction/algorithms/simulation/sidb/is_operational.hpp [hunk 3]
```diff
@@ -460,73 +460,26 @@ class is_operational_impl
...
-                            return true;
-                        }
-                    }
-                }
-            }
-        }
-
-        return false;
+        return std::any_of(input_bdl_wires.rbegin(), input_bdl_wires.rend(),
+                           [&, i = 0u](const auto& wire) mutable
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation/sidb/is_operational.hpp:464]: Global capture `[&]` should be avoided so as not to unintentionally capture variables. Rather, capture by name `[&ground_state, &current_input_index]`

File: include/fiction/algorithms/simulation/sidb/is_operational.hpp [hunk 4]
```diff
@@ -460,73 +460,26 @@ class is_operational_impl
...
+                           {
+                               const auto current_bit_set = (current_input_index & (uint64_t{1ull} << i++)) != 0ull;
+                               return std::any_of(wire.pairs.cbegin(), wire.pairs.cend(),
+                                                  [&](const auto& bdl)
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation/sidb/is_operational.hpp:468]: See above

File: include/fiction/algorithms/simulation/sidb/is_operational.hpp [hunk 5]
```diff
@@ -460,73 +460,26 @@ class is_operational_impl
...
-                        {
-                            return true;
-                        }
-                    }
-                }
-            }
-        }
-
-        return false;
+        return std::any_of(input_bdl_wires.rbegin(), input_bdl_wires.rend(),
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation/sidb/is_operational.hpp:463]:
```suggestion
        return std::any_of(input_bdl_wires.crbegin(), input_bdl_wires.crend(),
```

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: include/fiction/algorithms/physical_design/design_sidb_gates.hpp (hunk 2 above)
Comment (Drewniok) [reply include/fiction/algorithms/physical_design/design_sidb_gates.hpp:397]: oh, you are right! Many thanks!

File: include/fiction/algorithms/simulation/sidb/is_operational.hpp (hunk 3 above)
Comment (Drewniok) [reply include/fiction/algorithms/simulation/sidb/is_operational.hpp:464]: you are right! sorry for overseeing this.

**Review by marcelwa — APPROVED** [review]

**Pull request merged**